packages/flutter/lib/src/material/two_level_list.dart

```diff
@@ -154,11 +154,23 @@ class _TwoLevelSublistState extends State<TwoLevelSublist> {
 }
 
 class TwoLevelList extends StatelessWidget {
-  TwoLevelList({ Key key, this.items, this.type: MaterialListType.twoLine }) : super(key: key);
+  TwoLevelList({ Key key, this.scrollableKey, this.items, this.type: MaterialListType.twoLine }) : super(key: key);
 
   final List<Widget> items;
   final MaterialListType type;
+  final Key scrollableKey;
 
   @override
-  Widget build(BuildContext context) => new Block(children: items);
+  Widget build(BuildContext context) {
+    int itemIndex = 0;
+    List<Widget> itemsWithUniqueKeys = items?.map((Widget item) {
+      return new KeyedSubtree(
+        key: item.key != null ? new ValueKey<Key>(item.key) : new ValueKey<int>(itemIndex++),
```

Comment:
You probably want to increment itemIndex every trip through the loop, regardless of whether the item.key is null.

I wonder if we want to break this out into a helper function (maybe a static on KeyedSubtree) now that we've got this pattern in two places.  KeyedSubtree.ensureKeysForList(...) ?